src/metabase/models/serialization.clj

```diff
@@ -676,24 +690,37 @@
   (fn [ingested _]
     (ingested-model ingested)))
 
-(defn- load-by-spec [ingested]
-  (let [model-name (ingested-model ingested)
-        spec       (make-spec model-name)]
-    (when spec
-      (into (select-keys ingested (:copy spec))
-            (for [[k [_ser des]] (:transform spec)]
-              [k (des (get ingested k))])))))
+(defn- xform-by-spec [model-name ingested]
+  (binding [*current* ingested]
+    (let [spec (make-spec model-name nil)]
+      (when spec
+        (-> (select-keys ingested (:copy spec))
+            (into (for [[k transform] (:transform spec)
+                        :when (not (::nested transform))
+                        :let  [res ((:import transform) (get ingested k))]
+                        ;; do not try to insert nil values if transformer returns nothing
+                        :when res]
+                    [k res])))))))
+
+(defn- spec-post-process! [model-name ingested instance]
```

Comment:
"post-process" doesn't give the reader much info on what this does. You have to read the body of this function to understand it. How about naming this to be `load-nested!` or something that highlights that this is inserting nested entities?